test/string.js

```diff
@@ -2580,7 +2580,7 @@ describe('string', function () {
 
         it('validates combination of isoDate, min, max, regex and required', function (done) {
 
-            var rule = Joi.string().isoDate().min(17).max(23).regex(/Z$/).required();
+            var rule = Joi.string().isoDate().min(10).max(23).regex(/[^T]$|Z$|[+-][^a-zA-Z].*$/).required();
```

Comment:
This indeed changes the test too much, you basically removed the regexp from the equation.